SwiftUI/ReviewWQuestion-Calculator/ContentView.swift

```diff
@@ -5,15 +5,187 @@
+            return Color("signColorGray")
+        case .divide, .multiply, .minus, .plus, .equal:
+            return .orange
+        default:
+            return Color("digitColorGray")
+        }
+    }
+    
+    var textColor: Color {
+        switch self{
+        case .AC, .changeSign, .persent:
+            return Color(.black)
+        default:
+            return .white
+        }
+    }
+}
+    
+enum Operation {
+    case plus, minus, multiply, divide, none
+}
+
 struct ContentView: View {
+    
+    @State var currentValue: String = "0"
+    @State var nextValue: Int = 0
+//    @State var resultValue: Int = 0
+    @State var currentOperation: Operation = .none
+    
+    let digitButton: [[CalculatorButton]] = [
+        [.AC, .changeSign, .persent, .divide],
+        [.seven, .eight, .nine, .multiply],
+        [.four, .five, .six, .minus],
+        [.one, .two, .three, .plus],
+        [.zero, .zero, .dot, .equal]
+    ]
+    
+    let columns = [
+        //minimum값은 레이아웃의 최소 사이즈
+        GridItem(.adaptive(minimum: 80))
+    ]
+
     var body: some View {
-        Text("Hello, world!")
-            .padding()
+        ZStack{
+            Color.black
+                .ignoresSafeArea()
+            VStack{
+                HStack{
+                    Spacer()
+                    Text("\(currentValue)")
+                        .foregroundColor(.white)
+                        .font(.system(size: 90))
+                }
+                .padding(20)
+                
+                LazyVGrid(columns: columns, spacing: 15) {
+                    //rawvalue 공부할 것
+                    
+                    // 데이터가 각자를 식별할 수 있는 무언가를 주기 위해 id를 생성
+                    // id를 self로 줌으로써 ForEach 문 내에서 각 item의 id를 정해주도록 함. 이렇게 하면 Array 내부 요소가 가지고 있는 값 자체를 id로 가지게 해줌
+                    ForEach(digitButton, id: \.self) { row in
+                        ForEach(row, id: \.self) { item in
+                            Button(action: {
+                                switch item {
+                                case .plus, .minus, .multiply, .divide, .equal:
+                                    if item == .plus {
+                                        currentOperation = .plus
+                                        nextValue = Int(currentValue) ?? 0
+                                    } else if item == .minus {
+                                        currentOperation = .minus
+                                        nextValue = Int(currentValue) ?? 0
+                                    } else if item == .multiply {
+                                        currentOperation = .multiply
+                                        nextValue = Int(currentValue) ?? 0
+                                    } else if item == .divide {
+                                        currentOperation = .divide
+                                        nextValue = Int(currentValue) ?? 0
+                                    } else if item == .equal {
```

Comment:
```suggestion
                                case .plus:
                                        currentOperation = .plus
                                        nextValue = Int(currentValue) ?? 0
                                case .minus:
                                        currentOperation = .minus
                                        nextValue = Int(currentValue) ?? 0
                                case .multiply:
                                        currentOperation = .multiply
                                        nextValue = Int(currentValue) ?? 0
                                case .divide:
                                        currentOperation = .divide
                                        nextValue = Int(currentValue) ?? 0
                                case .equal:
```
if문 안쓰고 case를 저렇게 분기해서 썼어도 되지않았을까 싶습니다.